From 6241d84b52702ad9535781e0a8f4eea130323fd4 Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Thu, 18 Jul 2024 02:08:07 +0530
Subject: [PATCH 1/3] Fix the location of the Python include directory

CasADi's setup code used an incorrect location for the Python include
directory via the use of self.include_dirs[0], which isn't guaranteed
to return the correct interpreter used during compilation. This patch
replaces it with sysconfig.get_path('include').

This patch can be safely removed when the next release of CasADi comes
out, i.e., version 3.6.6. The relevant PR is 
https://github.com/casadi/casadi/pull/3759

---
 setup.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d7506513..42b8affb 100644
--- a/setup.py
+++ b/setup.py
@@ -6,6 +6,7 @@ import subprocess
 import pathlib
 import multiprocessing
 from io import open
+import sysconfig
 
 from setuptools import setup, Extension
 from setuptools.command.build_ext import build_ext
@@ -47,7 +48,7 @@ class CMakeBuild(build_ext):
             '-DSWIG_IMPORT=ON',
             '-DWITH_PYTHON=ON',
             '-DWITH_EXAMPLES=OFF',
-            '-DPYTHON_INCLUDE_DIR=' + self.include_dirs[0],
+            '-DPYTHON_INCLUDE_DIR=' + sysconfig.get_path('include'),
             '-DCMAKE_INSTALL_PREFIX=' + extdir,
             '-DWITH_SELFCONTAINED=ON',
             '-DWITH_DEEPBIND=ON'
-- 
2.39.3 (Apple Git-146)
